### DISCUSSION TITLE
Headers dictionary should be case insensitive

### DISCUSSION THREAD

**Discussion by grgouala** [dotnet/WatsonWebserver#104] [Ideas]
Hello,
I suggest to modify the Headers dictionary to a case insensitive one through the constructor with equality comparer set to _StringComparer.InvariantCultureIgnoreCase_.
Indeed, we can check the existence of a header by using the case unsensitive version of _HeaderExists_, but then we are not able to get the header value in the same manner.
That would be much easier and safer to remove completely any case sensitivity issues by making the Headers dictionary case sensitivity.

gregory

**Comment by jchristn** [comment]
Hi @grgouala I'd love to accept a PR on this if you're interested in doing it!  Cheers

**Reply by jchristn** [reply]
Hi @grgouala I made this change in v4.3.6, please let me know if any issues!  Thank you for your suggestion.

NuGet: https://www.nuget.org/packages/Watson/4.3.6
Commit: https://github.com/jchristn/WatsonWebserver/commit/176df4b032a68280860cf4c5339c9329a7dbb3ae
